src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -0,0 +1,257 @@
+        builder.append(processedSplitDate[DATE_YEAR_INDEX]);
+        return builder.toString();
+    }
+
+    /**
+     * Split the date String into integer array with a suitable separator for easy processing.
+     *
+     * @throws NumberFormatException if one segment of the date String cannot be parsed into an integer.
+     */
+    private static int[] processDate(String date) throws NumberFormatException {
+        requireNonNull(date);
+
+        String[] splitDate = getSplitDate(date);
+        int[] numericalSplitDate = new int[splitDate.length];
+
+        for (int i = 0; i < splitDate.length; i++) {
+            splitDate[i] = removeLeadingZeroes(splitDate[i]);
+            numericalSplitDate[i] = Integer.parseInt(splitDate[i]);
+        }
+
+        return numericalSplitDate;
+    }
+
+    /**
+     * Split the date String into a String array with a suitable separator.
+     */
+    private static String[] getSplitDate(String date) {
+        requireNonNull(date);
+
+        String[] splitDate;
+        if (date.contains(BIRTHDAY_DASH_SEPARATOR)) {
+            splitDate = date.split(BIRTHDAY_DASH_SEPARATOR);
+        } else if (date.contains(BIRTHDAY_FORWARD_SLASH_SEPARATOR)) {
+            splitDate = date.split(BIRTHDAY_FORWARD_SLASH_SEPARATOR);
+        } else {
+            splitDate = date.split(BIRTHDAY_DOT_SEPARATOR);
+        }
+        return splitDate;
+    }
+
+    /**
+     * Remove leading zeroes from each segment of the date to allow parsing using Integer.parseInt.
+     */
+    private static String removeLeadingZeroes(String toRemove) {
+        requireNonNull(toRemove);
+
+        while (!toRemove.isEmpty() && toRemove.startsWith("0")) {
+            toRemove = toRemove.substring(1);
+        }
+        return toRemove;
+    }
+
+    /**
+     * Returns true if the date is valid. A date is valid if it is possible for it to exist.
+     * E.g. 31/4/1999 is not a valid date because April only has 30 days.
+     */
+    public static boolean isValidBirthday(String birthday) {
+        requireNonNull(birthday);
+
+        if (birthday.isEmpty()) {
+            return true;
+        }
+
+        int[] processedSplitDate;
+
+        try {
+            processedSplitDate = processDate(birthday);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+
+        if (processedSplitDate.length != 3 || !isValidDate(processedSplitDate)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the date is valid. A date is valid if it is possible for it to exist.
+     */
+    private static boolean isValidDate(int[] processedSplitDate) {
+        requireNonNull(processedSplitDate);
+        if (isLeapYearDate(processedSplitDate)) {
+            return true;
+        }
+        if (isYearValid(processedSplitDate[DATE_YEAR_INDEX])
+                && isDayAndMonthValid(processedSplitDate[DATE_DAY_INDEX], processedSplitDate[DATE_MONTH_INDEX])) {
```

Comment:
If this is too long, maybe you can put it like this?
```
if (isYearValid(processedSplitDate[DATE_YEAR_INDEX])
    && isDayAndMonthValid(processedSplitDate[DATE_DAY_INDEX],
    processedSplitDate[DATE_MONTH_INDEX])) {
    return true;
}
```

Reply:
Actually I will just follow the first one you suggested. Seems better.

Reply:
idt this one should split. Its not too long and splitting in the middle of arguments is a bit confusing